### PR TITLE
docs: document reducedMotion API

### PR DESCRIPTION
**Pull request by parkerziegler**
This PR adds formal documentation for the new `reducedMotion` API!

The major changes include:

- Addition of a new page on Accessible Animations, with live code samples
- Addition of the `prefersReducedMotion` API and the `usePrefersReducedMotion` hook, with live code samples
- A few misc. stylistic improvements and resolution of development React errors.

### CODE REVIEW

File: docs/content/api.md [hunk 1]
```diff
@@ -13,23 +13,24 @@ This document contains all information on `renature`'s public facing API.
...
+| `from`                        | `{ [key: keyof React.CSSProperties]: any }`                                                        | The CSS property or properties you're animating **from**. Accepts an arbitrary number of key-value pairs.                                                                                                                                                     |
+| `to`                          | `{ [key: keyof React.CSSProperties]: any }`                                                        | The CSS property or properties you're animating **to**. Accepts an arbitrary number of key-value pairs.                                                                                                                                                       |
+| `reducedMotion`               | `{ from: { [key: keyof React.CSSProperties]: any }, to: { [key: keyof React.CSSProperties]: any }` | Optional. The animation configuration to use if the end user prefers reduced motion.                                                                                                                                                                          |
```
Comment (gingerrific) [inline docs/content/api.md]: Thoughts on adding a callout of the default behavior ~
```
Optional. The animation configuration to use if the end user prefers reduced motion. If no configuration is provided the animation will default to being "disabled" by setting both the "from" and "to" values to be the "to" values, meaning no transition will take place.
```
....but you know, with better words :P

Comment (gingerrific) [reply docs/content/api.md]: ah, I now see you also have the call out in the other docs - which is great, but probably wouldn't hurt to double up here

**Review by gingerrific — APPROVED** [review]
👍 nicely done!

File: docs/content/api.md (hunk 1 above)
Comment (parkerziegler) [reply docs/content/api.md]: Yep, agreed!

File: docs/content/api.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ This document contains all information on `renature`'s public facing API.
...
 | `to`                          | `{ [key: keyof React.CSSProperties]: any }`                                                        | The CSS property or properties you're animating **to**. Accepts an arbitrary number of key-value pairs.                                                                                                                                                       |
-| `reducedMotion`               | `{ from: { [key: keyof React.CSSProperties]: any }, to: { [key: keyof React.CSSProperties]: any }` | Optional. The animation configuration to use if the end user prefers reduced motion.                                                                                                                                                                          |
+| `reducedMotion`               | `{ from: { [key: keyof React.CSSProperties]: any }, to: { [key: keyof React.CSSProperties]: any }` | Optional. The animation configuration to use if the end user prefers reduced motion. If unspecified and the end user prefers reduced motion, the animating element will be set immediately to the `to` state to avoid potentially harmful animation.          |
```
Comment (gingerrific) [inline docs/content/api.md:20]: 🥳

**Review by upatel32 — APPROVED** [review]

**Pull request merged**